device/coordinate_manager.h

```diff
@@ -6,74 +6,161 @@
+        const tt_xy_pair& eth_grid_size,
+        const std::vector<tt_xy_pair>& eth_cores,
+        const tt_xy_pair& arc_grid_size,
+        const std::vector<tt_xy_pair>& arc_cores,
+        const tt_xy_pair& pcie_grid_size,
+        const std::vector<tt_xy_pair>& pcie_cores);
 
-    virtual tt_physical_coords to_physical_coords(tt_logical_coords logical_coords);
-    virtual tt_translated_coords to_translated_coords(tt_logical_coords logical_coords);
-    virtual tt_virtual_coords to_virtual_coords(tt_logical_coords logical_coords);
+    static std::unique_ptr<CoordinateManager> get_coordinate_manager(
+        tt::ARCH arch, const size_t tensix_harvesting_mask = 0, const size_t dram_harvesting_mask = 0);
 
-    virtual tt_logical_coords to_logical_coords(tt_physical_coords physical_coords);
-    virtual tt_virtual_coords to_virtual_coords(tt_physical_coords physical_coords);
-    virtual tt_translated_coords to_translated_coords(tt_physical_coords physical_coords);
+    CoordinateManager(CoordinateManager& other) = default;
 
-    virtual tt_logical_coords to_logical_coords(tt_virtual_coords virtual_coords);
-    virtual tt_physical_coords to_physical_coords(tt_virtual_coords virtual_coords);
-    virtual tt_translated_coords to_translated_coords(tt_virtual_coords virtual_coords);
+    CoreCoord to_physical(const CoreCoord core_coord);
+    CoreCoord to_logical(const CoreCoord core_coord);
+    CoreCoord to_virtual(const CoreCoord core_coord);
+    CoreCoord to_translated(const CoreCoord core_coord);
 
-    virtual tt_logical_coords to_logical_coords(tt_translated_coords translated_coords);
-    virtual tt_physical_coords to_physical_coords(tt_translated_coords translated_coords);
-    virtual tt_virtual_coords to_virtual_coords(tt_translated_coords translated_coords);
+    virtual ~CoordinateManager() = default;
 
-    static std::unique_ptr<CoordinateManager> get_coordinate_manager(
-        tt::ARCH arch,
-        const tt_xy_pair& worker_grid_size,
-        const std::vector<tt_xy_pair>& workers,
-        std::size_t harvesting_mask);
+protected:
+    void clear_tensix_harvesting_structures();
+    void clear_dram_harvesting_structures();
 
-    CoordinateManager(CoordinateManager& other) = default;
+    virtual void fill_tensix_logical_to_translated();
+    virtual void fill_eth_logical_to_translated();
+    virtual void fill_pcie_logical_to_translated();
 
-    virtual ~CoordinateManager() {}
+    std::map<size_t, size_t> physical_y_to_logical_y;
+    std::map<size_t, size_t> physical_x_to_logical_x;
 
-protected:
-    virtual void clear_harvesting_structures();
+    std::vector<size_t> logical_y_to_physical_y;
+    std::vector<size_t> logical_x_to_physical_x;
+
+    std::vector<size_t> logical_y_to_virtual_y;
+    std::vector<size_t> logical_x_to_virtual_x;
+
+    std::map<size_t, size_t> virtual_y_to_logical_y;
+    std::map<size_t, size_t> virtual_x_to_logical_x;
+
+    std::map<tt_xy_pair, CoreCoord> tensix_logical_to_translated;
```

Comment:
As we talked offline, I like the proposal for cache. Follow up issue to redesign that part, since I would like to push this PR to unblock clients (metal and lens) for the API work. 

On the second comment, I don't have strong opinions on this. I think yours and @abhullar-tt (and other metal folks) opinion is more important here, since it is more API design question. I can always forward the calls to already existing functions to_logical, to_physical, etc...

Reply:
From API point of view, scaling with adding additional entry to `enum` is easier than adding more functions... From user point of view, it is going to be explicit either way (either in function name or function parameter).

Reply:
I like the proposal to have `CoreCoord to(CoreCoord input, CoordSystem output) const;`

Under the hood it would be good to use some level of caching just to have fast translation.